src/app/ui-components/sidebar-right/sidebar-right.component.ts

```diff
@@ -57,8 +95,22 @@ export class SideBarRightComponent implements OnInit, OnDestroy {
                 }
             },
         );
-        this.selectedObservable?.subscribe((uuid?: string) => {
-            this.router.navigate([uuid ?? "."], {relativeTo: this.route});
+    }
+
+    toggleVoiceAssistant() {
+        const turnedOn = !this.voiceAssistantActivationToggle.value;
+        const nextState: VoiceAssistantState = {turnedOn, chatId: ""};
+        if (turnedOn) {
+            const match = RegExp(
+                `/voice-assistant/${CerebraRegex.UUID}/chat/(${CerebraRegex.UUID})`,
+            ).exec(this.router.url);
+            if (match) nextState.chatId = match[1];
+            else throw new Error("no chat selected");
+        }
+
+        this.vaState = turnedOn;
+        this.voiceAssistantService.setVoiceAssistantState(nextState).subscribe({
+            error: (error) => console.error(error),
         });
     }
```

Comment:
move this to the voice-assistant-chat component

Reply:
resolved